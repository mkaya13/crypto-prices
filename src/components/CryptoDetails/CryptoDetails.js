/* eslint-disable react/prop-types */
import React from 'react';

export default function CryptoDetails(props) {
  const { crypto } = props;
  const {
    name, symbol, marketCap, marketCapRank,
  } = crypto;

  return (
    <div>
      <li>
        <h1>{name}</h1>
        <h1>{symbol}</h1>
        <h1>{marketCap}</h1>
        <h1>{marketCapRank}</h1>
        <hr />
      </li>
    </div>
  );
}
