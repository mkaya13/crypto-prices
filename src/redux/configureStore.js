import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { cryptosReducer } from './cryptos/cryptos';
import { commentsByIdReducer } from './comments/commentsById';

const rootReducer = combineReducers({
  cryptos: cryptosReducer,
  commentsById: commentsByIdReducer,
});

export const store = configureStore({
  reducer: rootReducer,
});
export default { store };
