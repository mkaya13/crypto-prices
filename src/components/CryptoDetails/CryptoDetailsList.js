/* eslint-disable react/prop-types */
import React from 'react';
import { useParams } from 'react-router-dom';
import CryptoDetails from './CryptoDetails';

export const CryptoDetailsList = (props) => {
  const { cryptos } = props;
  const params = useParams();

  console.log(params.id);

  const myChoose = [...cryptos.filter((crypto) => crypto.id === params.id)];

  return (
    <ul className="each-crypto-table">
      {myChoose
        && myChoose.map((crypto) => (
          <CryptoDetails key={crypto.id} crypto={crypto} />
        ))}
    </ul>
  );
};

export default CryptoDetailsList;
