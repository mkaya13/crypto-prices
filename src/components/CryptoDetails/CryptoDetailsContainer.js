import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { CryptoDetailsList } from './CryptoDetailsList';
import { LoadCryptos } from '../../redux/cryptos/cryptos';

export const CryptoDetailsContainer = () => {
  const cryptos = useSelector((state) => state.cryptos);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(LoadCryptos());
  }, [dispatch]);

  return (
    <div className="crypto-items">
      <div className="main-page-title-details">
        <h2 className="crypto-detail-main-title">Details</h2>
      </div>
      <CryptoDetailsList cryptos={cryptos} />
    </div>
  );
};

export default { CryptoDetailsContainer };
