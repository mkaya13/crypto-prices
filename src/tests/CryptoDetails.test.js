import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import { BrowserRouter } from 'react-router-dom';

import CryptoDetails from '../components/CryptoDetails/CryptoDetails';

import { store } from '../redux/configureStore';

it('Renders CryptoDetails correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <CryptoDetails />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
