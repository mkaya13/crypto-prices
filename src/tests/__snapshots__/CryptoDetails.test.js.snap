// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders CryptoDetails correctly 1`] = `
<div
  className="detail-page-all"
>
  <div
    className="detail-page-main"
  >
    <img
      className="detail-page-crypto-logo"
    />
    <div
      className="detail-page-general-info"
    >
      <h2
        className="main-page-each-li-detail"
      >
        <span
          className="detail-feature-tag"
        >
          Name:
        </span>
         
      </h2>
      <h2
        className="main-page-each-li-detail"
      >
        <span
          className="detail-feature-tag"
        >
          Symbol:
        </span>
         
      </h2>
      <h2
        className="main-page-each-li-detail"
      >
        <span
          className="detail-feature-tag"
        >
          Price:
        </span>
         
         
        <span
          className="USD"
        >
          $
        </span>
      </h2>
    </div>
  </div>
  <ul
    className="detail-ul"
  >
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Market Cap:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Market Cap Rank:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Total Volume:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Highest 24h:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Lowest 24h
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Price Change 24h:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Market Cap Change
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Market Cap Change Percentage 24h:
      </span>
      <span>
         
        %
      </span>
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Circulating Supply:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Total Supply:
      </span>
      <span />
    </li>
    <li
      className="detail-each-li"
    >
      <span
        className="detail-feature-tag"
      >
        Max Supply:
      </span>
      <span />
    </li>
  </ul>
</div>
`;
